packages/flutter/lib/src/services/text_input.dart

```diff
@@ -856,6 +856,33 @@ abstract class TextInputClient {
   ///
   /// [TextInputClient] should cleanup its connection and finalize editing.
   void connectionClosed();
+
+  /// Requests that the client show the editing toolbar, for example when the
+  /// platform changes the selection through a non-flutter method such as
+  /// scribble.
+  void showToolbar();
+
+  /// UIKit has requested a text placeholder be added at the current selection
+  void insertTextPlaceholder(Size size);
```

Comment:
It does not appear that you can add multiple placeholders, that said, we do need to clean up the one in the current `TextInputClient` if the focus changes, since the `removeTextPlaceholder` call will get pushed to the new `TextInputClient`

Reply:
Oh, I forgot to answer your first question: this is another scribble feature where you can long press to give yourself extra space to write in the middle of some text. I don't know if it's used for anything else...the docs about it are pretty terse, and don't event mention the scribble use.